bleak/backends/scanner.py

```diff
@@ -208,8 +236,6 @@ def stop_if_detected(d: BLEDevice, ad: AdvertisementData):
                 await asyncio.wait_for(stop_scanning_event.wait(), timeout=timeout)
```

Comment:
We should be able to get rid of the `stop_scanning_event` altogether and do something like this instead.

```suggestion
                return await asyncio.wait_for(found_device_queue.get(), timeout=timeout)
```

Reply:
I tried that first, but there was an issue with it so I put the event back in... But now I don't remember the exact issue, but if the filter didn't trigger something went wrong.

Reply:
BTW: I won't be able to test/change anything until next weekend or later, so if you test the removal of the event and you're happy please just do it.

Reply:
ok, I went ahead and took care of it.